drivers/serial/uart_infineon_cat1.c

```diff
@@ -0,0 +1,467 @@
+/* Copyright (c) 2021 Cypress Semiconductor Corporation.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT infineon_mtbhal_uart
+
+/** @file
+ * @brief UART driver for Infineon CAT1 MCU family.
+ *
+ * Note:
+ * - Uart ASYNC functionality is not implemented in current
+ *   version of Uart CAT1 driver.
+ */
+#include <device.h>
+#include <drivers/uart.h>
+#include "cyhal_uart.h"
+
+/* Default UART interrupt priority*/
+#define UART_CAT1_INTERRUPT_PRIORITY      (5u)
+
+#define CYHAL_UART_ASYNC_TX_EVENTS        (CYHAL_UART_IRQ_TX_DONE | CYHAL_UART_IRQ_TX_ERROR)
+#define CYHAL_UART_ASYNC_RX_EVENTS        (CYHAL_UART_IRQ_RX_DONE | CYHAL_UART_IRQ_RX_ERROR)
+
+#define CYHAL_UART_IRQ_TX_EVENTS          (CYHAL_UART_IRQ_TX_DONE | CYHAL_UART_IRQ_TX_ERROR)
+#define CYHAL_UART_IRQ_RX_EVENTS          (CYHAL_UART_IRQ_RX_DONE | CYHAL_UART_IRQ_RX_ERROR)
+
+
+/* Data structure */
+struct uart_cat1_data {
+	cyhal_uart_t obj;                      /* UART CYHAL object */
+	struct uart_config cfg;
+
+    #if CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_callback_user_data_t irq_cb;   /* Interrupt Callback */
+	void *irq_cb_data;                      /* Interrupt Callback Arg */
+    #endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+
+    #if CONFIG_UART_ASYNC_API
```

Comment:
this PR has dozens of formatting issues (mixing tabs/spaces one of them), can you please review/fix all of them?